### ISSUE TITLE
"Ignore names found in names.txt" doesn't work in priority file generation

### ISSUE THREAD

**Issue by buster-blue** [mortii/anki-morphs#277]
### Describe the bug

I could be misunderstanding what this option is supposed to do, but it looks like names that are included in the names.txt file are still added to the priority file even when the option to ignore those names is enabled. I expected that they wouldn't show up because they're ignored, but all of the names still show up in the priority file anyway.

### Steps to reproduce the behavior

1. add names to names.txt
2. open the generator window and select text a folder of texts
3. enable the "Ignore names found in names.txt" option
4. generate the file
5. if the names are in those texts, they still get added to the priority file

### Expected behavior

I expect that names in the names.txt file won't be in the priority file, because they're supposed to be ignored.

### My AnkiMorphs settings

{
"algorithm_all_morphs_target_difference_weight": 10,
"algorithm_average_priority_all_morphs_weight": 0,
"algorithm_average_priority_learning_morphs_weight": 0,
"algorithm_learning_morphs_target_difference_weight": 10,
"algorithm_lower_target_all_morphs": 4,
"algorithm_lower_target_all_morphs_coefficient_a": 0.0,
"algorithm_lower_target_all_morphs_coefficient_b": 1.0,
"algorithm_lower_target_all_morphs_coefficient_c": 0.0,
"algorithm_lower_target_learning_morphs": 1,
"algorithm_lower_target_learning_morphs_coefficient_a": 6.0,
"algorithm_lower_target_learning_morphs_coefficient_b": 0.0,
"algorithm_lower_target_learning_morphs_coefficient_c": 0.0,
"algorithm_total_priority_all_morphs_weight": 1,
"algorithm_total_priority_learning_morphs_weight": 10,
"algorithm_total_priority_unknown_morphs_weight": 10,
"algorithm_upper_target_all_morphs": 6,
"algorithm_upper_target_all_morphs_coefficient_a": 1.0,
"algorithm_upper_target_all_morphs_coefficient_b": 0.0,
"algorithm_upper_target_all_morphs_coefficient_c": 0.0,
"algorithm_upper_target_learning_morphs": 2,
"algorithm_upper_target_learning_morphs_coefficient_a": 6.0,
"algorithm_upper_target_learning_morphs_coefficient_b": 0.0,
"algorithm_upper_target_learning_morphs_coefficient_c": 0.0,
"evaluate_morph_inflection": true,
"evaluate_morph_lemma": false,
"extra_difficulty": false,
"extra_fields_display_inflections": true,
"extra_fields_display_lemmas": false,
"extra_highlighted": false,
"extra_unknowns": true,
"extra_unknowns_count": true,
"filters": [
{
"extra_all_morphs": false,
"extra_all_morphs_count": false,
"extra_highlighted": false,
"extra_score": false,
"extra_score_terms": false,
"extra_study_morphs": true,
"extra_unknown_morphs": false,
"extra_unknown_morphs_count": true,
"field": "Expression",
"modify": true,
"morph_priority_selection": "priority-file.csv",
"morphemizer_description": "spaCy: ja_core_news_lg",
"note_type": "Japanese",
"read": true,
"tags": {
"exclude": [],
"include": []
}
},
{
"extra_all_morphs": false,
"extra_all_morphs_count": false,
"extra_highlighted": false,
"extra_score": false,
"extra_score_terms": false,
"extra_study_morphs": true,
"extra_unknown_morphs": false,
"extra_unknown_morphs_count": true,
"field": "Expression",
"modify": true,
"morph_priority_selection": "priority-file.csv",
"morphemizer_description": "spaCy: ja_core_news_lg",
"note_type": "Immersion Kit Japanese Sentence (my custom)",
"read": true,
"tags": {
"exclude": [],
"include": []
}
},
{
"extra_all_morphs": false,
"extra_all_morphs_count": false,
"extra_highlighted": false,
"extra_score": false,
"extra_score_terms": false,
"extra_study_morphs": true,
"extra_unknown_morphs": false,
"extra_unknown_morphs_count": true,
"field": "jpn",
"modify": true,
"morph_priority_selection": "priority-file.csv",
"morphemizer_description": "spaCy: ja_core_news_lg",
"note_type": "Tatoeba jpn",
"read": true,
"tags": {
"exclude": [],
"include": []
}
},
{
"extra_all_morphs": false,
"extra_all_morphs_count": false,
"extra_highlighted": false,
"extra_score": false,
"extra_score_terms": false,
"extra_study_morphs": true,
"extra_unknown_morphs": false,
"extra_unknown_morphs_count": true,
"field": "Expression",
"modify": true,
"morph_priority_selection": "priority-file.csv",
"morphemizer_description": "spaCy: ja_core_news_lg",
"note_type": "Japanese Core Anime Vocabulary",
"read": true,
"tags": {
"exclude": [],
"include": []
}
}
],
"hide_inflection_toolbar": false,
"hide_lemma_toolbar": false,
"hide_recalc_toolbar": false,
"interval_for_known_morphs": 5,
"parse_ignore_bracket_contents": false,
"parse_ignore_names_morphemizer": false,
"parse_ignore_names_textfile": false,
"parse_ignore_round_bracket_contents": false,
"parse_ignore_slim_round_bracket_contents": false,
"parse_ignore_suspended_cards_content": false,
"preprocess_ignore_bracket_contents": false,
"preprocess_ignore_names_morphemizer": true,
"preprocess_ignore_names_textfile": true,
"preprocess_ignore_round_bracket_contents": false,
"preprocess_ignore_slim_round_bracket_contents": false,
"preprocess_ignore_suspended_cards_content": false,
"read_known_morphs_folder": false,
"recalc_due_offset": 50000,
"recalc_interval_for_known": 5,
"recalc_move_known_new_cards_to_the_end": false,
"recalc_number_of_morphs_to_offset": 100,
"recalc_offset_new_cards": false,
"recalc_on_sync": false,
"recalc_read_known_morphs_folder": false,
"recalc_suspend_known_new_cards": true,
"recalc_toolbar_stats_use_known": false,
"recalc_toolbar_stats_use_seen": true,
"recalc_unknowns_field_shows_inflections": true,
"recalc_unknowns_field_shows_lemmas": false,
"shortcut_browse_all_same_unknown": "Shift+L",
"shortcut_browse_ready_same_unknown": "L",
"shortcut_browse_ready_same_unknown_lemma": "Ctrl+Shift+L",
"shortcut_frequency_file_generator": "Ctrl+Shift+F",
"shortcut_generators": "Ctrl+Shift+G",
"shortcut_known_morphs_exporter": "Ctrl+Shift+E",
"shortcut_learn_now": "Ctrl+Alt+N",
"shortcut_progression": "Ctrl+Alt+P",
"shortcut_readability_report_generator": "Ctrl+Shift+R",
"shortcut_recalc": "Ctrl+M",
"shortcut_set_known_and_skip": "K",
"shortcut_settings": "Ctrl+O",
"shortcut_view_morphemes": "Ctrl+Alt+V",
"skip_only_known_morphs_cards": false,
"skip_show_num_of_skipped_cards": true,
"skip_unknown_morph_seen_today_cards": false,
"tag_fresh": "am-fresh-morphs",
"tag_known_automatically": "am-known-automatically",
"tag_known_manually": "am-known-manually",
"tag_learn_card_now": "am-learn-card-now",
"tag_not_ready": "am-not-ready",
"tag_ready": "am-ready",
"toolbar_stats_use_known": false,
"toolbar_stats_use_seen": true
}

### My system

- Operating system: Fedora Linux 39
- Anki Ver. 24.06.3
- AnkiMorphs Ver. 3.0.0

### Additional context

I think this option still works when used in preprocessing for cards, but it's hard to check that. regardless, I tested with that option enabled and disabled, but the same problem still occurs with the generator option.

I also tested with a lemma only priority file, and with the occurrences column disabled, and recalculated every time just in case that might have something to do with it, but the names still show up.

**Comment by mortii** [comment]
Hmm, I can't immediately reproduce the problem:

![Screenshot from 2024-08-09 12-08-45](https://github.com/user-attachments/assets/225513cd-1390-4507-8811-3cfc2c24c000)
![Screenshot from 2024-08-09 12-08-51](https://github.com/user-attachments/assets/4b63d453-e3b4-4d5b-a82b-29dc2d023ffb)
![Screenshot from 2024-08-09 12-08-56](https://github.com/user-attachments/assets/c01fa1ae-e7a6-423c-84d2-a74441aa501c)

Could you provide some specific examples of names where this doesn't work?

**Comment by buster-blue** [comment]
I figured out the issue. I forgot to try something simple. I restarted Anki, and now it works. It looks like names.txt is only read on startup, and all of the names I was testing were names I recently added, so they weren't loaded when I generated the priority files. I checked the documentation page for it, and it didn't mention that you had to restart Anki for new names to be seen. I'd recommend putting some warning of that on the page, but that's a documentation issue not a bug, so I'll close this. thanks!
